week2/5.py

```diff
@@ -0,0 +1,23 @@
+# Write a Python script to check whether a given key already exists in a dictionary
+
+def checkKeyExist(dict,newKey,newValue):
+    # function takes 3 arguments a dictionary, new key and value
+    dictKeys = [] #an empty list to hold keys
+
+    # get keys from dictionnary and add to list
+    for key in car.keys():
+        dictKeys.append(key)
+
+    if newKey in dictKeys:
+        return f"key {newKey} already exist"
+    else:
+        dict[newKey] = newValue
+        return dict
+
+car = {
+  "brand": "Ford",
+  "model": "Mustang",
+  "year": 1964
+}
+
+print(checkKeyExist(car, "brand", "q"))
```

Comment:
There is something fishy here.You are using `car` inside the function, but it's defined outside. It runs, but is not a goo practice. The function is dependent on the outside scope which is bad. 
Also
```python
dictKeys = list(car.keys) # this works

# but also
print(if newKey in car) # this automatically looks at the keys of `car`

```

Reply:
ya i was wrong, it is 'dict'  not car